contracts/spend-limit/src/authenticator/handler/confirm_execution.rs

```diff
@@ -31,40 +33,49 @@ pub fn confirm_execution(
 
     let pre_exec_balances = pre_exec_balances.try_into()?;
     let post_exec_balances = post_exec_balances.try_into()?;
-    let spent_coins = calculate_spent_coins(pre_exec_balances, post_exec_balances)?;
-
-    let mut spending = SPENDINGS.load(deps.storage, spend_limit_key)?;
+    let mut spent_coins = calculate_spent_coins(pre_exec_balances, post_exec_balances)?;
 
-    let conf = PRICE_RESOLUTION_CONFIG.load(deps.storage)?;
+    // Get recent untracked spent fee, the latest fee is already captured in the balance difference, so we need to subtract it
+    let untracked_spent_fee = UNTRACKED_SPENT_FEES
+        .may_load(deps.storage, spend_limit_key)?
+        .unwrap_or_default()
+        .fee;
 
-    for coin in spent_coins.iter() {
-        // If the coin is not tracked, we don't count it towards the spending limit
-        let Some(price_info) = get_and_cache_price(
-            &PRICE_INFOS,
-            deps.branch(),
-            &conf,
-            env.block.time,
-            &coin.denom,
-        )?
-        else {
-            continue;
-        };
+    // add the untracked spent fee to the spent coins
+    for coin in untracked_spent_fee {
+        spent_coins.add(coin)?;
+    }
 
-        spending.spend(
-            coin.amount,
-            price_info.price,
-            params.limit,
-            &params.reset_period,
-            env.block.time,
-        )?;
+    // subtract the account spending fee from the spent coins
```

Comment:
```suggestion
    // To avoid double counting, we subtract the account spending fee from the spent coins. 
    // This is the fee for the current transaction and thus already captured by the difference in balances.
```